### ISSUE TITLE
StableDiffusionControlNetImg2ImgPipeline is missing from library

### ISSUE THREAD

**Issue by redromnon** [huggingface/diffusers#3737]
### Describe the bug

The `StableDiffusionControlNetImg2ImgPipeline` seems to be missing from the library when I try to import it.

I get this error:
 `ImportError: cannot import name 'StableDiffusionControlNetImg2ImgPipeline' from 'diffusers' 
(/usr/local/lib/python3.10/dist-packages/diffusers/__init__.py)`

To verify this, I ran `print(dir(diffusers))` and sure enough, `StableDiffusionControlNetImg2ImgPipeline` is absent.

```
['AltDiffusionImg2ImgPipeline', 'AltDiffusionPipeline', 'AudioDiffusionPipeline', 'AudioPipelineOutput', 'AutoencoderKL', 'ConfigMixin', 'ControlNetModel', 'CycleDiffusionPipeline', 'DDIMInverseScheduler', 'DDIMPipeline', 'DDIMScheduler', 'DDPMPipeline', 'DDPMScheduler', 'DEISMultistepScheduler', 'DPMSolverMultistepScheduler', 'DPMSolverSinglestepScheduler', 'DanceDiffusionPipeline', 'DiTPipeline', 'DiffusionPipeline', 'DummyObject', 'EMAModel', 'EulerAncestralDiscreteScheduler', 'EulerDiscreteScheduler', 'FlaxAutoencoderKL', 'FlaxDDIMScheduler', 'FlaxDDPMScheduler', 'FlaxDPMSolverMultistepScheduler', 'FlaxDiffusionPipeline', 'FlaxKarrasVeScheduler', 'FlaxLMSDiscreteScheduler', 'FlaxModelMixin', 'FlaxPNDMScheduler', 'FlaxSchedulerMixin', 'FlaxScoreSdeVeScheduler', 'FlaxStableDiffusionImg2ImgPipeline', 'FlaxStableDiffusionInpaintPipeline', 'FlaxStableDiffusionPipeline', 'FlaxUNet2DConditionModel', 'HeunDiscreteScheduler', 'IPNDMScheduler', 'ImagePipelineOutput', 'KDPM2AncestralDiscreteScheduler', 'KDPM2DiscreteScheduler', 'KarrasVePipeline', 'KarrasVeScheduler', 'LDMPipeline', 'LDMSuperResolutionPipeline', 'LDMTextToImagePipeline', 'LMSDiscreteScheduler', 'Mel', 'ModelMixin', 'OnnxRuntimeModel', 'OnnxStableDiffusionImg2ImgPipeline', 'OnnxStableDiffusionInpaintPipeline', 'OnnxStableDiffusionInpaintPipelineLegacy', 'OnnxStableDiffusionPipeline', 'OptionalDependencyNotAvailable', 'PNDMPipeline', 'PNDMScheduler', 'PaintByExamplePipeline', 'PriorTransformer', 'RePaintPipeline', 'RePaintScheduler', 'SchedulerMixin', 'ScoreSdeVePipeline', 'ScoreSdeVeScheduler', 'SemanticStableDiffusionPipeline', 'StableDiffusionAttendAndExcitePipeline', 'StableDiffusionControlNetPipeline', 'StableDiffusionDepth2ImgPipeline', 'StableDiffusionImageVariationPipeline', 'StableDiffusionImg2ImgPipeline', 'StableDiffusionInpaintPipeline', 'StableDiffusionInpaintPipelineLegacy', 'StableDiffusionInstructPix2PixPipeline', 'StableDiffusionKDiffusionPipeline', 'StableDiffusionLatentUpscalePipeline', 'StableDiffusionOnnxPipeline', 'StableDiffusionPanoramaPipeline', 'StableDiffusionPipeline', 'StableDiffusionPipelineSafe', 'StableDiffusionPix2PixZeroPipeline', 'StableDiffusionSAGPipeline', 'StableDiffusionUpscalePipeline', 'StableUnCLIPImg2ImgPipeline', 'StableUnCLIPPipeline', 'Transformer2DModel', 'UNet1DModel', 'UNet2DConditionModel', 'UNet2DModel', 'UnCLIPImageVariationPipeline', 'UnCLIPPipeline', 'UnCLIPScheduler', 'UniPCMultistepScheduler', 'VQDiffusionPipeline', 'VQDiffusionScheduler', 'VQModel', 'VersatileDiffusionDualGuidedPipeline', 'VersatileDiffusionImageVariationPipeline', 'VersatileDiffusionPipeline', 'VersatileDiffusionTextToImagePipeline', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', '__version__', 'configuration_utils', 'get_constant_schedule', 'get_constant_schedule_with_warmup', 'get_cosine_schedule_with_warmup', 'get_cosine_with_hard_restarts_schedule_with_warmup', 'get_linear_schedule_with_warmup', 'get_polynomial_decay_schedule_with_warmup', 'get_scheduler', 'is_flax_available', 'is_inflect_available', 'is_k_diffusion_available', 'is_k_diffusion_version', 'is_librosa_available', 'is_onnx_available', 'is_scipy_available', 'is_torch_available', 'is_transformers_available', 'is_transformers_version', 'is_unidecode_available', 'loaders', 'logging', 'models', 'optimization', 'pipeline_utils', 'pipelines', 'requires_backends', 'schedulers', 'training_utils', 'utils']
```

### Reproduction

Run 
```
from diffusers import StableDiffusionControlNetImg2ImgPipeline
```
and see the above error.

### System Info

- `diffusers` version: 0.14.0
- Platform: Linux-5.15.107+-x86_64-with-glibc2.31
- Python version: 3.10.12
- PyTorch version (GPU?): 2.0.1+cu118 (True)
- Huggingface_hub version: 0.15.1
- Transformers version: 4.30.1
- Accelerate version: 0.21.0.dev0
- xFormers version: 0.0.20
- Using GPU in script?: Nvidia T1

**Comment by mikegarts** [comment]
Hi @redromnon.

I see that the change allowing the direct usage via import was added only here https://github.com/huggingface/diffusers/blame/abbfe4b5b7a9508169e69476a951a1a6dd86c927/src/diffusers/__init__.py#L141 a month ago and I think it's available only since 0.17.0

Before that you could use StableDiffusionControlNetImg2ImgPipeline only through the community pipelines mechanism.

Please try a newer version of diffusers or load the pipeline as a community pipeline :)

Michael.

**Comment by redromnon** [comment]
Thanks for the clarification @mikegarts!

Maybe I should have checked by using the latest version of diffusers before creating an issue.
